### PR TITLE
chore(infra): migrate and adapt supernova from greenhouse extension to juno and fixed exampleapp

### PR DESCRIPTION
**Pull request by ArtieReus**
# Summary

Migrate and adapt supernova from greenhouse extension to juno. Fixed example app by adding missing dependencies.

# Changes Made

- Supernova migration from greenhouse-extension to juno
- Fixed exampleapp by adding missing dependencies

# Related Issues

- Issue 1: #298 

# Screenshots (if applicable)

# Testing Instructions

1. `npm i`
2. `npm run TASK`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: apps/supernova/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+COPY --from=build /app /usr/share/nginx/html
+
+ENV ENDPOINT=""
+ENV THEME=""
+ENV ENDPOINT="" 
+ENV FILTER_LABELS="" 
+ENV SILENCE_EXCLUDED_LABELS=""
+ENV SILENCE_TEMPLATES=""
+
```
Comment (edda) [inline apps/supernova/docker/Dockerfile:87]: `PREDEFINED_FILTERS` is missing here

File: apps/supernova/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    jq --arg v "\$value" ".\${field} = \\\$v" "\$APP_PROPS_FILE" > tmp.json && mv tmp.json "\$APP_PROPS_FILE"
+  fi
+}
+
+# Update fields if corresponding environment variables are set
+update_json_field "endpoint" "\$ENDPOINT"
+update_json_field "theme" "\$THEME"
+update_json_field "embedded" "\$EMBEDDED"
+update_json_field "silenceExcludedLabels" "\$SILENCE_EXCLUDED_LABELS"
+update_json_field "silenceTemplates" "\$SILENCE_TEMPLATES"
```
Comment (edda) [inline apps/supernova/docker/Dockerfile:121]: `PREDEFINED_FILTERS` missing

File: apps/supernova/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    "description": "Weekly maintenance window",
+    "editable_labels": ["app", "environment"],
+    "fixed_labels": {
+      "maintenance": "true",
+      "team": "devops"
+    },
+    "status": "active",
+    "title": "Weekly Maintenance"
+  }
+  ```
```
Comment (edda) [inline apps/supernova/docker/README.md:81]: `PREDEFINED_FILTERS` missing

File: apps/supernova/docker/appProps.sample.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "endpoint": "",
+  "theme": "theme-light",
+  "embedded": false,
+  "silenceExcludedLabels": [],
+  "silenceTemplates": {}
```
Comment (edda) [inline apps/supernova/docker/appProps.sample.json:6]: `predefinedFilters` missing

File: apps/supernova/package.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    },
+    "silenceTemplates": {
+      "value": null,
+      "type": "optional",
+      "description": "SilenceTemplates are pre-defined silence templates that can be used to scheduled Maintenance Windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title."
+    },
+    "predefinedFilters": {
+      "value": null,
+      "type": "optional",
+      "description": "PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, display name and matchers (map containing matchers as name and expression pairs)"
```
Comment (edda) [inline apps/supernova/package.json:88]: Need to fix this description
```suggestion
      "description": "PredefinedFilters are groups of filters comprised of a list of filter labels and a regex that is used to filter on the values of that label. They can be thought of as filter categories where the filter logic can be more complex than simple label+values pairs. Predefined filters are defined as an array of objects, where each object is a predefined filter. It contains the display name for the UI as well as a list of filter label + value regex pairs which are AND concatenated when they are being evaluated."
```

**Review by edda — CHANGES_REQUESTED** [review]

File: apps/supernova/docker/Dockerfile (hunk 1 above)
Comment (ArtieReus) [reply apps/supernova/docker/Dockerfile:87]: thx @edda!

File: apps/supernova/docker/Dockerfile (hunk 2 above)
Comment (ArtieReus) [reply apps/supernova/docker/Dockerfile:121]: thx @edda!

File: apps/supernova/docker/README.md (hunk 3 above)
Comment (ArtieReus) [reply apps/supernova/docker/README.md:81]: thx @edda!

File: apps/supernova/README.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+Sets the Alertmanager API Endpoint URL. Provide the full URL of the Alertmanager API endpoint to which the application will connect.
+
+To set the endpoint:
+
+1. Utilize the app prop `endpoint`, which is used during the setup of the script tag.
+
+### Predefined Filters
+
+Specifies filters to be applied in the UI for distinguishing between contexts by matching alerts with regular expressions. These filters are automatically loaded when the application starts. The format is a list of objects including name, displayname and matchers (containing keys corresponding value).
```
Comment (edda) [inline apps/supernova/README.md:159]: Use same description as in the package.json
```suggestion
PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, display name and matchers (map containing matchers as name and expression pairs).
```

File: apps/supernova/README.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+]
+```
+
+To set the predefined Filter:
+
+1. Utilize the app prop `predefinedFilters`, which is used during the setup of the script tag.
+
+### Silence Templates
+
+Defines pre-configured silences available in the schedule silence modal for setting up maintenance windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title.
```
Comment (edda) [inline apps/supernova/README.md:181]:
```suggestion
Defines pre-configured silences available in the schedule silence modal for scheduling future silences. For example when setting up maintenance windows where it is known that certain alerts will fire. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title.
```

**Review by edda — CHANGES_REQUESTED** [review]

**Review by edda — APPROVED** [review]

**Pull request merged**